python/hsfs/core/vector_server.py

```diff
@@ -232,45 +394,24 @@ def get_feature_vector(
                         self._complex_features, row._asdict()
                     )
                     serving_vector.update(result_dict)
-        # Add the passed features
-        serving_vector.update(passed_features)
-
-        # apply transformation functions
-        result_dict = self._apply_transformation(serving_vector)
-
-        vector = self._generate_vector(result_dict, allow_missing)
-
-        if return_type.lower() == "list":
-            return vector
-        elif return_type.lower() == "numpy":
-            return np.array(vector)
-        elif return_type.lower() == "pandas":
-            pandas_df = pd.DataFrame(vector).transpose()
-            pandas_df.columns = self._feature_vector_col_name
-            return pandas_df
-        else:
-            raise Exception(
-                "Unknown return type. Supported return types are 'list', 'pandas' and 'numpy'"
-            )
-
-    def get_feature_vectors(
-        self, entries, return_type=None, passed_features=[], allow_missing=False
-    ):
-        """Assembles serving vector from online feature store."""
+        return result_dict, serving_vector
```

Comment:
`result_dict` only stores the result of the last prepared statement. same for the batch case.

Reply:
I am not sure I understood. I just put this in a function to not duplicate the code. It must do the same as here https://github.com/logicalclocks/feature-store-api/blob/master/python/hsfs/core/vector_server.py#L248 

Am I missing something?

Reply:
[here](https://github.com/logicalclocks/feature-store-api/blob/master/python/hsfs/core/vector_server.py#L248)
`result_dict` store the value of transformed `serving_vector`
`result_dict = self._apply_transformation(serving_vector)`

but in the code above 
```
for row in result_proxy:
                    result_dict = self.deserialize_complex_features(
                        self._complex_features, row._asdict()
                    )
                    serving_vector.update(result_dict)
```
the last assignment to result_dict is inside a for loop.

Reply:
yes but `_vector_result` is just copy of from [here](https://github.com/logicalclocks/feature-store-api/blob/master/python/hsfs/core/vector_server.py#L214) to [here](https://github.com/logicalclocks/feature-store-api/blob/master/python/hsfs/core/vector_server.py#L243). That was the whole idea. I am still unable to spot difference